backend/middleware/authMiddleware.ts

```diff
@@ -1,11 +1,12 @@
 import jwt from 'jsonwebtoken'
 import { type Response, type NextFunction } from 'express'
 import asyncHandler from 'express-async-handler'
-import User from '../models/userModel'
+import { UserModel } from '../models/userModel'
+import { config } from '../config'
 
 let JWT_SECRET: string
-if (process.env.JWT_SECRET) {
-  JWT_SECRET = process.env.JWT_SECRET
+if (config.jwtSecret) {
+  JWT_SECRET = config.jwtSecret
 } else {
   throw new Error('JWT_SECRET environment variable is not set')
```

Comment:
Unnecessary validation, you already did that in the config file.

Reply:
you're right, removed this part!